src/generators.h

```diff
@@ -60,6 +60,7 @@ enum struct DeviceType {
   CUDA,
   DML,
   WEBGPU,
+  QNN_WITH_SHARED_MEMORY,
```

Comment:
Will there be a future distinction between `QNN` and `QNN_WITH_SHARED_MEMORY`? If not, should we consider renaming this to just `QNN`?

Reply:
I think we could. @edgchen1 should we rename to QNN?

Reply:
sure